### PR TITLE
Simplify code, fix navigation, set up pages rendering using slugs

### PR DESCRIPTION
**Pull request by muhabalwan**

### CODE REVIEW

File: components/Header.tsx [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+
+const Header = ({navList}: IHeader) => {
     return (
         <>
         <Head>
             <title>Home Page</title>
             <meta name="myProfile" content="web development, freelance, muhab alwan profile, javascript developer, frontend" />
             <link rel="icon" href="/favicon.ico" />
         </Head>
+        <Navigation navList={navList} />
```
Comment (muhabalwan) [inline components/Header.tsx:19]: Navigation should be in the Header component

File: components/Navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import navStyles from '../styles/Navigation.module.css';
```
Comment (muhabalwan) [inline components/Navigation.tsx:1]: Rename Nav.tsx to Navigation.tsx 
Remove uneeded functionality

File: pages/PageContainer.tsx [hunk 3]
```diff
@@ -1,44 +1,46 @@
...
 
-    const createMarkup = (html: string) => {
-        return {__html: html};
-    }
+  const createMarkup = (html: string) => {
+    return { __html: html };
+  }
 
-    return (
-    <Layout pages={props.pages} getSelectedLink={getSelectedLink} navList={navList}>
```
Comment (muhabalwan) [inline pages/PageContainer.tsx:37]: Layout is not needed, deleted Layout component

File: pages/PageContainer.tsx [hunk 4]
```diff
@@ -1,44 +1,46 @@
...
 
-    const page = props?.pages?.find(page => {
-        return page.slug === props.pageSlug
-      });
+  useEffect(() => {
+    const pageFromSlug = getPageFromSlug(pageSlug);
+    setSelecetedPage(pageFromSlug);
+  });
 
-    const getSelectedLink = (selectedLink: ISelectedLink) => {
```
Comment (muhabalwan) [inline pages/PageContainer.tsx:28]: This is way too complicated I removed it, we can just get the selectedLink from Routes check pageSlug

File: pages/[pageSlug].tsx [hunk 5]
```diff
@@ -4,15 +4,20 @@ import { getContentSettings } from './api/contentSettings';
...
 
 export default function DynamicPage(props: IPageProps) {
   const router = useRouter();
   const { pageSlug } = router.query;
-  // TODO: how can we make this better
   if (typeof pageSlug === "string") {
     return (
-      <PageContainer pageSlug={pageSlug} pages={props.pages} navList={props.navList}/>
+      <>
+        <Header navList={props.navList} />
```
Comment (muhabalwan) [inline pages/[pageSlug].tsx:17]: I think the code should be as simple as this no more

**Comment by AhmadYoussef** [comment]
I will check it later then i will call you

**Pull request merged**